### DISCUSSION TITLE
Clear rebranding for v2.0 -- stackoverflow tag for sqlalchemy-2.0 and maybe sqlalchemy-1.4

### DISCUSSION THREAD

**Discussion by taigit** [sqlalchemy/sqlalchemy#8157] [Ideas]
Context: I'm getting back to coding my sql alchemy project and the different version of sql alchemy are getting me confused.

Key usage method: to look up a sql alchemy question, I use google search:  sqlalchemy query scalars

Problem: with the different versions of sql alchemy: v1.3, different subversion of v1.4, and v2.0.   Now, my google search is less effective because the search results don't differentiate between the versions.

Suggestion: the launch of  sqlalchemy 2.0 should rebrand to make it clear to google search that this advice is for v2.0 so we can use google search for documentation and help.  eg: similar to the rebranding of python-2.x vs python-3.x

Concrete suggestion: there should be a stackoverflow tag for  sqlalchemy-2.0

In fact, there should be  a stackoverflow tag for  sqlalchemy-1.4-transitional also since there seems to be foundational code that does not work for v1.3, but works for v1.4.24.  In particular, this example  https://docs.sqlalchemy.org/en/14/orm/quickstart.html#simple-select  requires v1.4.24 because of  session.scalars() is introduced there.

**Comment by CaselIT** [comment]
Hi,

Not sure if sqlalchemy can create tags in stackoverflow or how the work in the specific, but it makes sense to me to have a 2 tag to indicate the answers with the new syntax.

Cc @gordthompson

**Comment by zzzeek** [comment]
IMO stackoverflow doesn't really scale well for time, because whether or not you have new tags, SO is loaded with old obsolete answers for all kinds of projects because it has no facility for dealing with obsolete versions of software.     it's a problem for them, our own solution here is dont use stackoverflow, use our own discussions up above.

**Comment by gordthompson** [comment]
I've thought about adding a sqlalchemy-2.0 tag on SO but, as @zzzeek says, I'm not sure how much it would help.

They are also (still) thinking about the issue of outdated answers, e.g.,

https://meta.stackoverflow.com/questions/405302/introducing-outdated-answers-project

but it will probably be difficult to overcome the sheer volume of old answers, especially since they've had years to accumulate upvotes.

**Comment by taigit** [comment]
As an end user (basic level), google search for sqlalchemy answers is 10x the most efficient way to find answers currently.  It is a key part of the sqlalchemy ecosystem for end users...!

Stackoverflow does have lots of old answers, but version-specific tags is the SO solution for that (eg: python-3.x ).

Then, end users like myself will use Google search (not SO search) to find sqlalchemy-2.0 -- and I think sqlalchemy is important enough that google will make sure it does a good job at v2.0 answers if end users put that in the google search query string.

Google will also find v2.0 answers in this discussion group too -- that's why end users like google search so much!

As for the old answers, they are actually needed for those using v1.0 sqlalchemy.

Thus, there is a need to differentiate answers for the various versions and do it across multiple answer locations.  Google will do its magic to direct end users to the right type of answer, but **it would be great to give Google (and answer creators) advice in labeling the different types of sqlalchemy answers...**  If you have a different solution for Stackoverflow, that's great.  We just need some SO solution since lots of v2.0 answers will end up there.

I think the better the end user experience is for using v2.0 the faster the adoption of v2.0 will be and the more impact sqlalchemy will have on computation in general!

**Comment by taigit** [comment]
Wow, this summary poster about how to use v2.0 (and difference from 1.x)  is so clear and concise!  It's great!  Thank you!

https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#migration-orm-usage

**Reply by CaselIT** [reply]
thanks!

**Reply by taigit** [reply]
Congrats on getting v2.0 out!
I'm checking it out now.

One of the first places I checked is this:
https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#migration-orm-usage

This is a formatting comment, but an important one: that table is cut off on Chrome and Firefox browsers so i cannot see much of the 2nd column...  and i don't see a workaround (other than scaling to .75x )

**Reply by taigit** [reply]
BTW, since i'm asking about that table, let me ask what I was researching.  

Can I (and should I) use the 2.0 style for my queries?   Can I use the 2.0 style in my v1.4 version of sqlalchemy?

**Reply by zzzeek** [reply]
you can use 2.0 style in 1.4, certainly, for select(), update(), and delete() statements.    

I see the issue with the table and I dont have a fix other than we should format the code to be really narrow...  hmmmm @CaselIT  is this our code formatter doing this?  uh oh !

**Reply by zzzeek** [reply]
cool I can do "format: off" in that block.  will fix

**Reply by zzzeek** [reply]
I've pushed a change that narrows those examples way down.   I think a horizontal formatted table wasn't the best format for this, as they look funny being so narrowly formatted as well, but it will get us through.

**Reply by CaselIT** [reply]
> cool I can do "format: off" in that block. will fix

I half though about removing this, but I thought it may be useful for special cases. Maybe we could instruct black to use a different line length via directive? (probably not worth the effort)

**Reply by zzzeek** [reply]
yeah I had thought of that, special black line length.  but if you look at what I did, it probably isn't black-compatible anyway because i did only 2 spaces indent inside for line continuation.

**Reply by CaselIT** [reply]
I think we can keep it like this since that seems the only place where it's needed.

**Comment by davidism** [comment]
Version tags are almost never used correctly on Stack Overflow, especially the `[python-3.*]` tags. Users know what version they're using, but can never identify whether an issue is actually about that version and not their general code (it's always the later). Please don't create a new tag. We already follow the `[sqlalchemy]` tag. If you think the issue is with a specific version, write that in the question. Google will continue to work just fine.
